sharding/xorValidatorShuffler.go

```diff
@@ -170,7 +178,8 @@ func shuffleNodesIntraShards(
 
 	waiting, stillRemainingInLeaving = removeFromWaiting(waiting, stillRemainingInLeaving)
 
-	for shard, validators := range eligible {
+	for shard := uint32(0); shard < uint32(len(eligible)-1); shard++ {
```

Comment:
same thing as my previous comment regarding iterating on a map

Reply:
good catch. there are multiple places in the application where we do a range over the maps and expect that the number of shards is equal to the len of the maps.

Reply:
not the case any more